### PR TITLE
fix(getFuturesChangePositionTxData): Replace constant with calculated…

### PR DESCRIPTION
**Pull request by hochbruj**
… price

### CODE REVIEW

File: src/services/futures/trade.ts [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
+  market: string,
+  pool: Pool
+): Promise<string> {
+  const futuresMarket = new ethers.Contract(
+    market,
+    ISynthetixFuturesMarketV2.abi,
+    pool.signer
+  );
+  const fillPrice = await futuresMarket.fillPrice(amount);
+  const adjustmentFactor = ethers.BigNumber.from(amount).lt(0) ? 995 : 1005;
```
Comment (itsermin) [inline src/services/futures/trade.ts:16]: Maybe a comment would be helpful eg
"Allows for +-0.5% price movements on the desired fill price"

File: src/services/futures/trade.ts [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
+
   return new ethers.utils.Interface(
     ISynthetixFuturesMarketV2.abi
   ).encodeFunctionData("submitOffchainDelayedOrderWithTracking", [
     amount,
-    PRICE_IMPACT_DELTA,
+    fillPrice.price.mul(adjustmentFactor).div(1000),
```
Comment (itsermin) [inline src/services/futures/trade.ts:22]: Just for ledgibility, we could have this defined as a constant `desiredFillPrice` further up.

**Review by itsermin — APPROVED** [review]
Just a couple of minor comments

File: src/services/futures/trade.ts (hunk 1 above)
Comment (hochbruj) [reply src/services/futures/trade.ts:17]: added comment

File: src/services/futures/trade.ts (hunk 2 above)
Comment (hochbruj) [reply src/services/futures/trade.ts]: added constant

**Pull request merged**